node/v2/handler.js

```diff
@@ -183,14 +183,21 @@ TChannelV2Handler.prototype.handleInitResponse = function handleInitResponse(res
 
 TChannelV2Handler.prototype.handleCallRequest = function handleCallRequest(reqFrame) {
     var self = this;
+
     if (self.remoteName === null) {
-        return self.errorEvent.emit(self, errors.CallReqBeforeInitReqError());
+        self.errorEvent.emit(self, errors.CallReqBeforeInitReqError());
+        return;
     }
 
     var req = self.buildInRequest(reqFrame);
 
-    if (self.incomingRequestInvalid(reqFrame, req)) {
-        return null;
+    var err = self.checkCallReqFrame(reqFrame);
+    if (err) {
+        req.res = self.buildOutResponse(req);
+        self.sendErrorFrame(
+            req.res, 'ProtocolError', err.message
```

Comment:
We can switch here on the type of `err` and send an appropriate error type.

We can use `Errors.classify(err) || 'ProtocolError'`

Let's verify that the arg1 over length error is properly classified.